src/std/misc/with-id-test.ss

```diff
@@ -0,0 +1,39 @@
+(export with-id-test)
+
+(import
+  :gerbil/gambit/exceptions
+  :std/srfi/13
+  :std/misc/number
+  :std/misc/string
+  :std/sugar
+  :std/stxutil
+  :std/test
+  ./with-id)
+
+(def with-id-test
+  (test-suite "test suite for std/misc/with-id"
+    (test-case "with-id, defining variables"
+      (def mem (make-vector 5 0))
+      (defrule (defvar name n)
+        (with-id defvar ((@ #'name "@") (get #'name) (set #'name "-set!"))
```

Comment:
the correct context here is `name`.

Reply:
Aren't the two contexts the same at that point? To quote the non-president "What difference, at this point, does it make?"

Reply:
Followed your advice.